### PR TITLE
Hair 43 combine users

### PR DESCRIPTION
**Pull request by cynthiaeddy**
as discussed in today's meeting, this branch is a combination of:
1. logging out toggles navbar to signup/login.
2. search is now home page (page name is still Search).
3. signup page is for both user and business owner.

### CODE REVIEW

File: models/business.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ const businessSchema = new mongoose.Schema({
   state: { type: String },
   country: { type: String },
   mainPicture: { type: String },
+  email: { type: String, required: true, unique: true },
   status: { type: Number, default: 1 },
-  email: { type: String },
-  features: [{ type: ObjectId, ref: 'features' }],
-  services: [{ type: ObjectId, ref: 'services' }]
+  features: { type: ObjectId, ref: 'features' },
```
Comment (CodeMeKathy) [inline models/business.js:19]: @cynthiaeddy -> Hi! Can you please elaborate on why you made this modification?

File: models/business.js [hunk 2]
```diff
@@ -14,10 +14,10 @@ const businessSchema = new mongoose.Schema({
...
+  services: { type: ObjectId, ref: 'services' },
```
Comment (CodeMeKathy) [inline models/business.js:20]: @cynthiaeddy -> Hi! Can you please elaborate on why you made this modification?

File: models/business.js (hunk 1 above)
Comment (cynthiaeddy) [reply models/business.js:19]: @CodeMeKathy Hi! I don't know why that showed up - here's what I have on my local (screenshot). I'd say delete it.
Screen Shot 2021-11-13 at 9 53 35 AM

File: client/src/pages/BusinessProfile.js [hunk 3]
```diff
@@ -4,13 +4,13 @@ import React, { useState, useContext, useEffect } from 'react'
...
 import ContentHeader from '../components/ContentHeader'
-import { AuthContext } from '../contexts/GlobalContext'
+import { AppContext } from '../contexts/AppContext'
 
 const BusinessProfile = () => {
   const history = useHistory()
 
   // Import User State Object from Context
-  const { userState, setUserState } = useContext(AuthContext)
+  const { user, setUser } = useContext(AppContext)
```
Comment (CodeMeKathy) [inline client/src/pages/BusinessProfile.js:13]: Hi @cynthiaeddy, the `userId` is not accessible from context during the `saveNewBusiness` function and is not saved as a new document in the businesses collection. See images with a newly added user, Ben Walker with Walker Cutz, as compared to previous user Billy Stone with Stone Cutz below.

image

**Review by CodeMeKathy — CHANGES_REQUESTED** [review]
Hi @cynthiaeddy, the `userId` should be added to the business profile during creation and as part of the new document in the businesses collection. With this PR that is no longer happening. Please see images with a newly added user, Ben Walker with Walker Cutz, as compared to previous user Billy Stone with Stone Cutz below for details.

File: models/business.js (hunk 1 above)
Comment (CodeMeKathy) [reply models/business.js:19]: Hi @cynthiaeddy! Your picture aligns with the committed changes; the `businessSchema` features property is no longer set to an array of `ObjectId`s. As a business can have more than one feature and service we need to be prepared to store more than one. This was initially accomplished via an array.

My initial comment was an attempt to understand better the reason behind the change here and to services. I do not believe I can delete it. You would need to undo/revert this change and commit the modifications to the PR.

**Pull request closed without merging**